### PR TITLE
[cling] Expose ROOTLIBDIR to cling::DynamicLibraryManager:

### PR DESCRIPTION
**Pull request by Axel-Naumann**
DynamicLibraryManager only looks at LD_LIBRARY_PATH and friends, but for install / prefix
builds that might not point to the ROOT library dir. Make it explicit that the
DynamicLibraryManager is supposed to look at the directory containing ROOT libraries.
Fixes install-with-prefix builds with RPATH, that now do not need a fake LD_LIBRARY_PATH
anymore.

### CODE REVIEW

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -1437,8 +1437,14 @@ TCling::TCling(const char *name, const char *title, const char* const argv[])
...
 
-   // We are set up.
-   EnableAutoLoading();
+   if (!fromRootCling) {
+      // Make sure cling looks into ROOT's libdir, even if not part of LD_LIBRARY_PATH
+      // e.g. because of an RPATH build.
+      fInterpreter->getDynamicLibraryManager()->addSearchPath(TROOT::GetLibDir().Data());
+
+      // We are set up.
+      EnableAutoLoading();
```
Comment (vgvassilev) [inline core/metacling/src/TCling.cxx]: We have an early exit if we are in rootcling mode: https://github.com/root-project/root/blob/master/core/metacling/src/TCling.cxx#L2931

Comment (Axel-Naumann) [reply core/metacling/src/TCling.cxx]: Thanks, well spotted! As I favor readability over having to remember or looking up I'll just leave it if you don't mind - it shouldn't be expensive.

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

File: core/metacling/src/TCling.cxx (hunk 1 above)
Comment (pcanal) [reply core/metacling/src/TCling.cxx]: Wouldn't a comment be as readable as adding 'redundant' calls?

Comment (Axel-Naumann) [reply core/metacling/src/TCling.cxx]: If both of you insist... I do find a comment less visible than code, but I did what the two of you suggested.

**Comment by Axel-Naumann** [comment]
@phsft-bot build just on windows10/cxx14

**Pull request merged**